ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java

```diff
@@ -146,15 +176,18 @@ private ParseReportingUrl(String urlAllowList) {
                 attributes.get(Attribute.GEO_SUBDIVISION1));
             urlParser.addQueryParam(ParsedReportingUrl.PARAM_OS_FAMILY,
                 getOsParam(attributes.get(Attribute.USER_AGENT_OS)));
-            urlParser.addQueryParam(ParsedReportingUrl.PARAM_FORM_FACTOR, "desktop");
+            urlParser.addQueryParam(ParsedReportingUrl.PARAM_FORM_FACTOR, interaction.form());
 
             urlParser.addQueryParam(ParsedReportingUrl.PARAM_DMA_CODE,
                 message.getAttribute(Attribute.GEO_DMA_CODE));
           }
 
           // We only add these dimensions for topsites clicks, not quicksuggest per
+          // We are also limiting this to desktop.
           // https://bugzilla.mozilla.org/show_bug.cgi?id=1738974
-          if (message.getAttribute(Attribute.DOCUMENT_TYPE).equals("topsites-click")) {
+          if (interaction.form().equals(SponsoredInteraction.FORM_DESKTOP)
+             && interaction.source().equals(SponsoredInteraction.SOURCE_TOPSITE)
+             && interaction.interaction().equals(SponsoredInteraction.INTERACTION_CLICK)) {
```

Comment:
```suggestion
          if (SponsoredInteraction.FORM_DESKTOP.equals(interaction.form())
             && SponsoredInteraction.SOURCE_TOPSITE.equals(interaction.source())
             && SponsoredInteraction.INTERACTION_CLICK.equals(interaction.interaction())) {
```

To avoid potential NPE, it's safer to call `STRING_CONSTANT.equals(value)` than the other way around.